### PR TITLE
feat(providers): add webex oauth

### PR DESCRIPTION
**Pull request by askmrsinh**
## ☕️ Reasoning

Adds Webex Oauth Provider, Similarweb rank: [2,441](https://www.similarweb.com/website/webex.com/#overview). Used by many private/public orgs (including universities, goverment - state and federal, etc.).

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

 
N/A

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/core/src/providers/webex.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function Webex<P extends WebexProfile>(
+  config: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  const apiBaseUrl = config?.apiBaseUrl ?? 'https://webexapis.com/v1';
```
Comment (askmrsinh) [inline packages/core/src/providers/webex.ts:77]: PS: This line does not pass typechcking.

File: packages/core/src/providers/webex.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function Webex<P extends WebexProfile>(
+  config: OAuthUserConfig<P>
```
Comment (ndom91) [inline packages/core/src/providers/webex.ts:75]: Extending this is what most providers do, something like:

```suggestion
  config: OAuthUserConfig<P> & { baseUrl: string }
```

File: packages/core/src/providers/webex.ts (hunk 1 above)
Comment (askmrsinh) [reply packages/core/src/providers/webex.ts:78]: Please see https://github.com/nextauthjs/next-auth/pull/9653#discussion_r1460702670.

File: packages/core/src/providers/webex.ts (hunk 2 above)
Comment (askmrsinh) [reply packages/core/src/providers/webex.ts]: Done.

File: packages/core/src/providers/webex.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  return {
+    id: 'webex',
+    name: 'Webex',
+    type: 'oauth',
+    authorization: {
+      url: apiBaseUrl + '/authorize',
+      params: { scope: 'spark:kms spark:people_read' }
+    },
+    token: apiBaseUrl + '/access_token',
+    userinfo: apiBaseUrl + '/people/me',
```
Comment (ndom91) [inline packages/core/src/providers/webex.ts:89]: Tiny nit on these, can you change them to string interpolation? i.e. `${apiBaseUrl}/people/me`? :pray:

Comment (askmrsinh) [reply packages/core/src/providers/webex.ts]: Done!

**Comment by askmrsinh** [comment]
@ndom91, @ThangHuuVu - if no more changes are required, can this be merged?

**Comment by zoneix** [comment]
Hi folks! The PR looks great and will greatly benefit our  various Webex related projects. Would be nice if we could get this merged!

File: packages/core/src/providers/webex.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+/**
+ * The returned user profile from Webex when using the profile callback.
+ *
+ * Please refer to {@link https://developer.webex.com/docs/api/v1/people/get-my-own-details|People - Get My Own Details}
+ * on Webex Developer portal for additional fields. Returned fields may vary depending on the user's role, the OAuth
+ * integration's scope, and the organization the OAuth integration belongs to.
+ */
+export interface WebexProfile extends Record<string, any> {
+  id: string;
+  emails: [string, ...string[]];
```
Comment (ndom91) [inline packages/core/src/providers/webex.ts:22]: Is this supposed to be a string or an array of strings? i.e. `string | string[]`?

Comment (askmrsinh) [reply packages/core/src/providers/webex.ts:22]: The later one. It's an array of string, with at least one element.

Comment (askmrsinh) [reply packages/core/src/providers/webex.ts]: I have updated it to `string[]`.

**Review by ndom91 — APPROVED** [review]

**Pull request merged**